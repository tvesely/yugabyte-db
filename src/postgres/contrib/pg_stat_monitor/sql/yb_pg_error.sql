CREATE EXTENSION pg_stat_monitor;
SELECT pg_stat_monitor_reset();
SELECT 1/0;   -- divide by zero
SELECT * FROM unknown; -- unknown table
-- https://github.com/yugabyte/yugabyte-db/issues/11801
-- TODO: works with postgres and not with YB
-- ELECET * FROM unknown; -- syntax error

do $$
BEGIN
RAISE WARNING 'warning message';
END $$;

-- TODO: SQL code sometimes did not work in mac and alma platforms for YB
-- SELECT query, elevel, sqlcode, message FROM pg_stat_monitor ORDER BY query COLLATE "C",elevel;
SELECT query, elevel, message FROM pg_stat_monitor ORDER BY query COLLATE "C",elevel;
SELECT pg_stat_monitor_reset();
DROP EXTENSION pg_stat_monitor;
