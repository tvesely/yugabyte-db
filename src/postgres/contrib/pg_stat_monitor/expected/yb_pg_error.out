CREATE EXTENSION pg_stat_monitor;
SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset
-----------------------

(1 row)

SELECT 1/0;   -- divide by zero
ERROR:  division by zero
SELECT * FROM unknown; -- unknown table
ERROR:  relation "unknown" does not exist
LINE 1: SELECT * FROM unknown;
                      ^
-- https://github.com/yugabyte/yugabyte-db/issues/11801
-- TODO: works with postgres and not with YB
-- ELECET * FROM unknown; -- syntax error
do $$
BEGIN
RAISE WARNING 'warning message';
END $$;
WARNING:  warning message
-- TODO: SQL code sometimes did not work in mac and alma platforms for YB
-- SELECT query, elevel, sqlcode, message FROM pg_stat_monitor ORDER BY query COLLATE "C",elevel;
SELECT query, elevel, message FROM pg_stat_monitor ORDER BY query COLLATE "C",elevel;
                                         query                                         | elevel |              message
---------------------------------------------------------------------------------------+--------+-----------------------------------
 SELECT * FROM unknown;                                                                |     20 | relation "unknown" does not exist
 SELECT 1/0;                                                                           |     20 | division by zero
 SELECT pg_stat_monitor_reset();                                                       |      0 |
 SELECT query, elevel, message FROM pg_stat_monitor ORDER BY query COLLATE "C",elevel; |      0 |
 do $$                                                                                +|     19 | warning message
 BEGIN                                                                                +|        |
 RAISE WARNING 'warning message';                                                     +|        |
 END $$;                                                                               |        |
(5 rows)

SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset
-----------------------

(1 row)

DROP EXTENSION pg_stat_monitor;
